src/http/client.cc

```diff
@@ -19,10 +19,116 @@
  * Copyright (C) 2022 Scylladb, Ltd.
  */
 
+#include <seastar/core/loop.hh>
+#include <seastar/core/when_all.hh>
 #include <seastar/http/client.hh>
+#include <seastar/http/request.hh>
+#include <seastar/http/reply.hh>
+#include <seastar/http/response_parser.hh>
+#include <seastar/http/internal/content_source.hh>
 
 namespace seastar {
 namespace http {
 
+client::client(connected_socket&& fd)
+        : _fd(std::move(fd))
+        , _read_buf(_fd.input())
+        , _write_buf(_fd.output())
+{
+}
+
+future<> client::write_body(request& req) {
+    if (req.body_writer) {
+        return req.body_writer(internal::make_http_chunked_output_stream(_write_buf)).then([this] {
+            return _write_buf.write("0\r\n\r\n");
+        });
+    } else if (!req.content.empty()) {
+        return _write_buf.write(req.content);
+    } else {
+        return make_ready_future<>();
+    }
+}
+
+future<std::optional<reply>> client::maybe_wait_for_continue(request& req) {
+    if (req.get_header("Expect") == "") {
```

Comment:
I guess you assume that Expect with anything is "Expect: 100-continue"?

Reply:
Yes :) This is what httpd assumes to, so I just made this the simplest way possible